### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Monoidal.Free.Basic

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
TODO:

- [ ] name in comments
- [ ] check naming in general
- [ ] two unsolved goals in the last def `project`

**Comment by mo271** [comment]
https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!4.232808.20CategoryTheory.2EMonoidal.2EFree.2EBasic

File: Mathlib/CategoryTheory/Monoidal/Free/Basic.lean [hunk 1]
```diff
@@ -46,302 +46,291 @@ Given a type `C`, the free monoidal category over `C` has as objects formal expr
...
-  forall {C : Type.{u2}} {D : Type.{u3}} [_inst_1 : CategoryTheory.Category.{u1, u3} D] [_inst_2 : CategoryTheory.MonoidalCategory.{u1, u3} D _inst_1] (f : C -> D) {X : CategoryTheory.FreeMonoidalCategory.{u2} C} {Y : CategoryTheory.FreeMonoidalCategory.{u2} C}, (CategoryTheory.FreeMonoidalCategory.Hom.{u2} C X Y) -> (Quiver.Hom.{succ u1, u3} D (CategoryTheory.CategoryStruct.toQuiver.{u1, u3} D (CategoryTheory.Category.toCategoryStruct.{u1, u3} D _inst_1)) (CategoryTheory.FreeMonoidalCategory.projectObj.{u1, u2, u3} C D _inst_1 _inst_2 f X) (CategoryTheory.FreeMonoidalCategory.projectObj.{u1, u2, u3} C D _inst_1 _inst_2 f Y))
-but is expected to have type
-  PUnit.{max (max (succ (succ u1)) (succ (succ u2))) (succ (succ u3))}
-Case conversion may be inaccurate. Consider using '#align category_theory.free_monoidal_category.project_map_aux CategoryTheory.FreeMonoidalCategory.projectMapAuxₓ'. -/
-/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
-/-- Auxiliary definition for `free_monoidal_category.project`. -/
-@[simp]
+/-- Auxiliary definition for `FreeMonoidalCategory.project`. -/
+-- Porting note: here `@[simp]` would generates a panic in
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Monoidal/Free/Basic.lean:262]:
```suggestion
-- Porting note: here `@[simp]` generates a panic in
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**